### PR TITLE
Default the SWRA to abortJobsOnTerminate = true.

### PR DESCRIPTION
**Pull request by kshakir**

### CODE REVIEW

**Comment by kshakir** [comment]
To reviewers: review the commits separately. The second is a typo fix.

File: engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala [hunk 1]
```diff
@@ -105,28 +111,46 @@ class WorkflowManagerActor(config: Config,
   private val tag = self.path.name
 
   private var abortingWorkflowToReplyTo = Map.empty[WorkflowId, ActorRef]
+  private var shutdownHookThreadOption: Option[ShutdownHookThread] = None
 
   override def preStart(): Unit = {
+    tryRemoveShutdownHook()
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: why is this needed here?

**Comment by mcovarr** [comment]
:+1:

File: README.md [hunk 2]
```diff
@@ -328,6 +328,8 @@ system {
 
 Or, via `-Dsystem.abort-jobs-on-terminate=true` command line option.
 
+By default, this value is false when running `java -jar cromwell.jar server`, and true when running `java -jar cromwell.jar run <wdl> <inputs>`.
```
Comment (cjllanwarne) [inline README.md]: Also CHANGELOG.MD

File: core/src/main/resources/reference.conf [hunk 3]
```diff
@@ -56,7 +56,7 @@ akka {
 
 system {
   # If 'true', a SIGINT will trigger Cromwell to attempt to abort all currently running jobs before exiting
-  abort-jobs-on-terminate = false
+  #abort-jobs-on-terminate = false
```
Comment (cjllanwarne) [inline core/src/main/resources/reference.conf]: Space: `# abort-jobs-...`

File: engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala [hunk 4]
```diff
@@ -94,8 +98,10 @@ class WorkflowManagerActor(config: Config,
            jobStoreActor: ActorRef,
            callCacheReadActor: ActorRef,
            jobTokenDispenserActor: ActorRef,
-           backendSingletonCollection: BackendSingletonCollection) = this(
-    ConfigFactory.load, workflowStore, serviceRegistryActor, workflowLogCopyRouter, jobStoreActor, callCacheReadActor, jobTokenDispenserActor, backendSingletonCollection)
+           backendSingletonCollection: BackendSingletonCollection,
+           abortJobsOnTerminate: Boolean) = this(
+    ConfigFactory.load, workflowStore, serviceRegistryActor, workflowLogCopyRouter, jobStoreActor, callCacheReadActor,
```
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: ToL: Since this entire list is triplicated, can we DRY it at all?

(e.g. wrap it up in a `WmaParameters` case class?)

File: engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala [hunk 5]
```diff
@@ -105,28 +111,45 @@ class WorkflowManagerActor(config: Config,
...
       }
+      shutdownHookThreadOption = Option(shutdownHookThread)
+    }
+  }
+
+  private def tryRemoveShutdownHook() = {
+    try {
+      shutdownHookThreadOption.foreach(_.remove())
+    } catch {
+      case _: Exception => /* ignore */
```
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: `log.error("The world is ending and it's all your fault!")`?

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala [hunk 6]
```diff
@@ -390,11 +390,12 @@ final case class WorkflowExecutionActor(workflowId: WorkflowId,
...
   when(WorkflowExecutionAbortingState) {
-    case Event(AbortedResponse(jobKey), stateData) =>
-      workflowLogger.info(s"$tag job aborted: ${jobKey.tag}")
+    case Event(response: BackendJobExecutionResponse, stateData) =>
+      val jobKey = response.jobKey
+      workflowLogger.info(s"$tag job exited with ${response.getClass.getSimpleName}: ${jobKey.tag}")
       val newStateData = stateData.removeBackendJobExecutionActor(jobKey)
       if (newStateData.backendJobExecutionActors.isEmpty) {
-        workflowLogger.info(s"$tag all jobs aborted")
+        workflowLogger.info(s"$tag all jobs exited")
```
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala]: ToL: I think you were ok before... this is `when(WorkflowExecutionAbortingState)`

File: engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala [hunk 7]
```diff
@@ -105,28 +111,45 @@ class WorkflowManagerActor(config: Config,
...
 
   override def preStart(): Unit = {
     addShutdownHook()
     // Starts the workflow polling cycle
     self ! RetrieveNewWorkflows
   }
 
+  override def postStop() = {
+    // If the actor is stopping, then there's nothing to wait for later. Remove the shutdown hook.
+    tryRemoveShutdownHook()
```
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: ToL:

Hmm, in this case either we've already stopped everything (so the shutdown hook would be a no-op) or there are jobs left running (and we'd want to run the shutdown hook, assuming it was added)

So, I think we'd prefer to not have this line?

**Comment by cjllanwarne** [comment]
👍 pending changes

File: README.md (hunk 2 above)
Comment (kshakir) [inline README.md]: Done

File: core/src/main/resources/reference.conf (hunk 3 above)
Comment (kshakir) [inline core/src/main/resources/reference.conf]: Leaving, it's not comment text, but a commented-out value. The lack of space matches other commented-out values in the file.

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala (hunk 6 above)
Comment (kshakir) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala]: Discussed. With `system.abort-jobs-on-terminate=true`, a Control-C at just the right time can cause some jobs to exit with "success" messages. But depending on the rest of the application configuration, the stored "RUNNING" jobs might just re-run when cromwell restarts.

**Comment by kshakir** [comment]
PR comments addressed. Will squash and merge in a bit unless there are additional comments to address.
